### PR TITLE
Feat: Add social links to the profile showcase

### PR DESCRIPTION
**Pull request by MishkaRogachev**
For https://github.com/status-im/status-desktop/issues/13425

Important changes:
- [x] Limit every profile showcase section with 20 entries
- [x] Add social links to the profile showcase
- [x] Migrate existing data
- [ ] Support existing status-go api for backward compatibility
- [ ] Remove unused social links code

### CODE REVIEW

**Review by igor-sirotin — APPROVED** [review]
👍 
I like when things are implemented in the way that adding another property is easy!

File: protocol/identity/profile_showcase.go [hunk 1]
```diff
@@ -6,6 +6,10 @@ var ErrorNoAccountProvidedWithTokenOrCollectible = errors.New("no account provid
 var ErrorDublicateAccountAddress = errors.New("duplicate account address")
 var ErrorAccountVisibilityLowerThanCollectible = errors.New("account visibility lower than collectible")
 
+var ErrorExceedMaxProfileEntriesLimit = errors.New("exeed maximum profile entries limit per section")
+
+const maxProfileEntriesLimit = 20
```
Comment (saledjenic) [inline protocol/identity/profile_showcase.go]: @MishkaRogachev we need this limitation on the UI side (as well).

File: protocol/migrations/sqlite/1708440786_profile_showcase_social_links.up.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    sort_order INT DEFAULT 0,
+    contact_id TEXT NOT NULL,
+    PRIMARY KEY (contact_id, text, url)
+);
+CREATE INDEX profile_showcase_social_links_contact_id ON profile_showcase_social_links_contacts (contact_id);
+
+-- Copy existing social links to a new table
+INSERT INTO profile_showcase_social_links_preferences (text, url, sort_order)
+    SELECT text, url, position
+    FROM profile_social_links;
```
Comment (saledjenic) [inline protocol/migrations/sqlite/1708440786_profile_showcase_social_links.up.sql:23]: @MishkaRogachev do we want to delete `profile_social_links` table?

File: protocol/persistence_profile_showcase.go [hunk 3]
```diff
@@ -528,13 +581,48 @@ func (db sqlitePersistence) getProfileShowcaseUnverifiedTokensContact(tx *sql.Tx
...
+		err := rows.Scan(
+			&link.URL,
+			&link.Text,
+			&link.Order)
+		if err != nil {
+			return nil, err
+		}
+
+		links = append(links, link)
+	}
```
Comment (saledjenic) [inline protocol/persistence_profile_showcase.go:624]:
```suggestion
        }
        err = rows.Err()
	if err != nil {
		return nil, err
	}
```

**Review by saledjenic — COMMENTED** [review]
I don't see related changes for backing up social links along with the profile data? There are still "old" social links, do you plan to remove them (clean the code) later?

File: protocol/identity/profile_showcase.go (hunk 1 above)
Comment (MishkaRogachev) [reply protocol/identity/profile_showcase.go]: Yes. For now i guess we use 20 for all profile entries

File: protocol/migrations/sqlite/1708440786_profile_showcase_social_links.up.sql (hunk 2 above)
Comment (MishkaRogachev) [reply protocol/migrations/sqlite/1708440786_profile_showcase_social_links.up.sql:23]: Yes, but not in this PR. First the [UI](https://github.com/status-im/status-desktop/issues/13606) should switch to a new backend, and here is an [issue](https://github.com/status-im/status-go/issues/4792) for it

Comment (MishkaRogachev) [reply protocol/migrations/sqlite/1708440786_profile_showcase_social_links.up.sql:23]: Yes, but not in this PR. First the [UI](https://github.com/status-im/status-desktop/issues/13606) should switch to a new backend, and here is an [issue](https://github.com/status-im/status-go/issues/4792) for it.

**Review by saledjenic — APPROVED** [review]

File: protocol/migrations/sqlite/1708440786_profile_showcase_social_links.up.sql (hunk 2 above)
Comment (MishkaRogachev) [reply protocol/migrations/sqlite/1708440786_profile_showcase_social_links.up.sql:23]: Yes, but not in this PR. First the [UI](https://github.com/status-im/status-desktop/issues/13606) should switch to a new backend, and here is an [issue](https://github.com/status-im/status-go/issues/4792) for it.

**Comment by MishkaRogachev** [comment]
Question with backup, investigating

File: protocol/identity/profile_showcase.go (hunk 1 above)
Comment (MishkaRogachev) [reply protocol/identity/profile_showcase.go]: UPD: 20 for social links, 100 for others

**Pull request merged**